src/Engine.sol

```diff
@@ -431,12 +473,12 @@ contract Engine is IEngine, Multicallable, EIP712, EIP7412, ERC2771Context {
 
             // ensure position exists; reduce only orders cannot increase position size
             if (positionSize == 0) {
-                return (retOrder, 0, 0);
+                return (retOrder, 0);
```

Comment:
the keeper gets the fee here right? 

so if someone submits a conditional order for a no longer valid position, the executor gets to keep the fee?

Reply:
> the keeper gets the fee here right?

Yes, the executor fee is imposed prior to this. However, the Synthetix imposed fee will be zero since no order will be committed

> so if someone submits a conditional order for a no longer valid position, the executor gets to keep the fee?

The simple answer is Yes. The reason is because the keeper did work to execute an order that should have been canceled by the user (i.e. invalidate the nonce and thus `canExecute` returns `false` preventing wasted gas)